etsin_finder/frontend/js/components/qvain/files/dataCatalog.jsx

```diff
@@ -66,16 +81,46 @@ class DataCatalog extends Component {
           options={availableOptions}
           onChange={(selection) => {
             setDataCatalog(selection.value)
-            this.setState({ errorMessage: undefined })
+            this.setState({
+              errorMessage: undefined,
+              fileOrigin: selection.label,
+            })
           }}
           onBlur={this.handleOnBlur}
           attributes={{ placeholder: 'qvain.files.dataCatalog.placeholder' }}
           isDisabled={(selected.length > 0) || (original !== undefined) || isPas}
         />
+        {
+          (this.state.fileOrigin === 'IDA' && original === undefined) && (
+          <DoiSelectionContainer>
+            <Checkbox
+              id="doiSelector"
+              onChange={this.handleDoiCheckboxChange}
+              disabled={(this.state.fileOrigin !== 'IDA' || original !== undefined)}
+              defaultChecked={this.state.useDoi || ((original !== undefined) && (dataCatalog === 'urn:nbn:fi:att:data-catalog-ida'))}
+            />
```

Comment:
I think in future we could make a list of catalogs and refer to the list instead of hard-coding the names, especially since we might get more cases like Jyväskylä and the amount of catalogs will increase and their names might change. But no need to make changes now.

Reply:
Good suggestion for the future